File: aoj/itp1/js/2-d-circle-in-rectangle.js



const rl = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("", (args) => {
  // 長方形右上頂点座標(W, H)、円の中心座標(x, y)、半径r
  const [W, H, x, y, r] = args.split(" ").map(Number);

  if (x + r > W) {
    console.log("No");
  } else if (x - r < 0) {
    console.log("No");
  } else if (y + r > H) {
    console.log("No");
  } else if (y - r < 0) {
    console.log("No");
  } else {
    console.log("Yes");
  }
  rl.close();
});

File: aoj/itp1/py/2-d-circle-in-rectangle.py
# 長方形右上頂点座標(W, H)、円の中心座標(x, y)、半径r
W, H, x, y, r = map(int, input().split(" "))

# 左にはみ出る
if x - r < 0:
  print("No")
  exit(0)

# 右にはみ出る
if x + r > W:
  print("No")
  exit(0)

# 下にはみ出る
if y - r < 0:
  print("No")
  exit(0)

# 上にはみ出る
if y + r > H:
  print("No")
  exit(0)

print("Yes")





